torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -83,8 +82,9 @@ std::shared_ptr<FutureMessage> sendPythonRemoteCall(
 
   // set forceGradRecording to true as even if the args does not contain any
   // tensor, the return value might still contain tensors.
+  auto agent = RpcAgent::getDefaultRpcAgent();
   return torch::distributed::autograd::sendMessageWithAutograd(
```

Comment:
should this instead take a `std::shared_ptr&`?

Reply:
I guess it want to save the trouble of checking nullptr.
If agent is a nullptr. I guess *agent should raise an exception?
I don't want to change about it in this PR, unless I have clear nullptr checking guideline in mind.